apps/src/aichat/views/ChatEventsList.tsx

```diff
@@ -21,36 +23,98 @@ const ChatEventsList: React.FunctionComponent<ChatEventsListProps> = ({
   events,
   isTeacherView,
 }) => {
+  const [inProgrammaticScroll, setInProgrammaticScroll] = useState(false);
+  const [showScrollToBottom, setShowScrollToBottom] = useState(true);
   const {isWaitingForChatResponse} = useAppSelector(state => state.aichat);
 
   // Compare the chat events  as a string since the object reference will change on every update.
   // This way we will only scroll when the contents of the events have changed.
   const eventsString = JSON.stringify(events);
   const conversationContainerRef = useRef<HTMLDivElement>(null);
 
-  useEffect(() => {
+  const scrollToBottom = () => {
     if (conversationContainerRef.current) {
-      conversationContainerRef.current.scrollTo({
-        top: conversationContainerRef.current.scrollHeight,
-        behavior: 'smooth',
-      });
+      setShowScrollToBottom(false);
+
+      // On clear chat, we don't want to scroll to the bottom.
+      if (eventsString !== '[]') {
+        setInProgrammaticScroll(true);
```

Comment:
I wanted to prevent the button from showing as we do "automated" scrolls when a new message is received. This involved "turning off" the scroll event handler while those "programatic scrolls" are in flight, and setting a setInterval to repeatedly check until we're at the end of the scroll, at which point we turn the event handler back on. [There is a "scrollend" event](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollend_event) that could be used rather than polling for the scroll to complete, but it is only newer browser versions (and not at all on Safari).